docs/specs/2fa-grace-period.md

```diff
@@ -20,29 +20,55 @@ For new users, the grace period starts from the time the user is created, wherea
 
 ### Global Grace Period
 
-Global grace period can be enabled for forcing users to enroll in 2FA upon login.
+Global grace period can be enabled for forcing users to enroll in 2FA upon login instead of blocking them.
 
 ```yaml
 authentication:
   secondary_authentication_mode: "required"
-  secondary_authentication_rollout:
-    global_grace_period_enabled_since: "2021-01-01T00:00:00Z"
-    global_grace_period_days: 30
+  secondary_authentication_grace_period:
+    enabled: true
+    end_at: "2021-01-01T00:00:00Z"
 ```
 
 ### Per-user Grace Period
 
-Regardless of the global grace period, a per-user grace period can be granted for 10 days by default, or a custom duration can be specified:
-
-```yaml
-authentication:
-  secondary_authentication_mode: "required"
-  secondary_authentication_rollout:
-    per_user_grace_period_days: 10
+Regardless of global grace period, specific users can be granted grace period through Admin Portal / GraphQL API.
+
+```gql
+type User {
+  id: ID!
+  # ...
+
+  mfaGracePeriodEndAt: DateTime
+}
+
+type GrantMFAGracePeriodInput {
```

Comment:
Replace the word "grant" with "set", in input, payload, mutation name, etc.

Reply:
Updated.